src/components/Feed.svelte

```diff
@@ -1,154 +1,41 @@
-		['limit', slotLimit.toString()],
-		['feedType', 'global']
-	]);
-	let paramsGlobalOnly = new URLSearchParams([
-		['postId', ''],
-		['limit', slotLimit.toString()]
-	]);
-
-	async function loadMorePosts() {
-		if (loginToken !== '' && value === 0) {
-			paramsChangeable.set('postId', feedData.pagination.lastPostId.toString());
-			paramsChangeable.set('feedType', feedType);
-		} else if (loginToken !== '' && value === 1) {
-			paramsChangeable.set('postId', feedData.pagination.lastPostId.toString());
-		} else {
-			paramsGlobalOnly.set('postId', feedData.pagination.lastPostId.toString());
-		}
-
-		let params = loginToken === '' ? paramsGlobalOnly : paramsChangeable;
-		const url: string = serverUrl + '/feed?' + params;
-		try {
-			let response = await fetch(url, {
-				mode: 'cors',
-				method: 'GET',
-				headers: {
-					Authorization: 'Bearer ' + get(token)
-				}
-			});
-			statusCode = response.status;
-			if (statusCode === 200) {
-				const result = await response.json();
-				if (result.records.length === 0) {
-					hasMorePosts = false;
-				} else {
-					maxPostCounter += result.records.length;
-					posts = posts.concat(result.records);
-					feedData.records = feedData.records.concat(result.records);
-					feedData.pagination.lastPostId = result.pagination.lastPostId;
-				}
-			}
-		} catch (error) {
-			toastStore.clear();
-			toastStore.trigger(createToast('Internal Server Error! Please try again later!', 'error'));
-		}
-	}
-
-	function setFeedType() {
-		if (loginToken !== '' && value === 1) {
-			paramsChangeable.set('feedType', 'personal');
-			feedType = 'personal';
-		} else {
-			paramsChangeable.set('feedType', 'global');
-			feedType = 'global';
-		}
-	}
-
-	async function onChange() {
-		setFeedType();
-		fetchMatchingFeed(false);
-	}
-
-	async function fetchMatchingFeed(isGlobal: boolean) {
-		let params = isGlobal ? paramsGlobalOnly : paramsChangeable;
-		const url: string = serverUrl + '/feed?' + params;
-		try {
-			let response = await fetch(url, {
-				mode: 'cors',
-				method: 'GET',
-				headers: {
-					Authorization: 'Bearer ' + get(token)
-				}
-			});
-			statusCode = response.status;
-
-			if (statusCode === 200) {
-				const result = await response.json();
-				posts = result.records;
-				maxPostCounter += posts.length;
-				feedData = result;
-			} else if (statusCode !== 200 && statusCode !== 500) {
-				//Fehler-Handling von 400 Bad Request
-				// error objekt auslesen, wie result s.o. customError.message
-				// copy from register/verify
-				posts = [];
-				toastStore.clear();
-				toastStore.trigger(createToast($t('toast.somethingWrong'), 'error'));
-			}
-		} catch (error) {
-			toastStore.clear();
-			toastStore.trigger(createToast($t('toast.internalError'), 'error'));
-		}
+	async function onLoadMorePosts() {
+		const loginToken = get(token);
+		const serverUrl = get(serverURL);
+		await loadMorePosts(loginToken, serverUrl, toastStore, pageType);
 	}
 
 	onMount(async () => {
```

Comment:
man braucht keine onMount hier, die daten fetchen wir uns jeweils in den +page.svelte dateien und übergeben sie nur an die komponenten, dafür sind sie ja da